### DISCUSSION TITLE
Extend or override vite vue plugin

### DISCUSSION THREAD

**Discussion by joneath** [storybookjs/storybook#22879] [Help] [answered]
With the release of Vue 3.3 I've adopted the experimental [`defineModel`](https://github.com/vuejs/rfcs/discussions/503) but I can't seem to enable this compiler flag in storybook's vite config. When importing the vue plugin from `@vitejs/plugin-vue` and including it in storybook's `viteFinal` plugins array like so:
```
vuePlugin({
  script: {
    defineModel: true,
    propsDestructure: true,
  },
})
```

I get the following error when booting storybook:
`[vite] Internal server error: At least one <template> or <script> is required in a single file component.`

I assume this is because when defining my own `vuePlugin` I am overriding the default one which is correctly configured. Any thoughts on how to either extend this default vue vite plugin or correctly override it with the experimental compiler flags?

**Comment by mneciunas** [comment, marked as answer]
I've had the same problem when using mergeConfig
```
async viteFinal(viteConfig) {
  return mergeConfig(viteConfig, {
    plugins: [
      vue(
        script: {
          defineModel: true,
        },
      )
    ],
  }
}
```

the solution was to remove vue plugin before merging like this:
`viteConfig.plugins = viteConfig.plugins.filter(p => p.name !== 'vite:vue');`
in the end i've got this to work:
```
async viteFinal(viteConfig) {
  viteConfig.plugins = viteConfig.plugins.filter(p => p.name !== 'vite:vue');

  return mergeConfig(viteConfig, {
    plugins: [
      vue(
        script: {
          defineModel: true,
        },
      )
    ],
  }
}
```

**Reply by joneath** [reply]
Woohoo, this works for me as well. Thank you!

**Reply by IanVS** [reply]
@joneath do you not have your own vite config for your app?

**Reply by joneath** [reply]
Yes I have to separate them out as my app is a Nuxt app

**Reply by IanVS** [reply]
The other thing you can do, then, is create a vite config just for storybook, and use `builder.options.viteConfigPath` in `main.ts` to point to that.  If you do, then storybook won't inject its own default vite plugins that you have to remove in `viteFinal`.

**Reply by joevaugh4n** [reply]
Huge thanks for helping out here, @mneciunas and @IanVS! 🎉

**Comment by IanVS** [comment]
Storybook will use the plugin defined in your vite config, so you should only need to add/configure it there.  Unless you want a different configuration between your app and your Storybook?

**Reply by usrrname** [reply]
This is exacly the situation i've landed in as folks at my work would prefer not to mess with the root level nuxt's `vite.config.js`
